tests/lint/test_namespace_packages.py

```diff
@@ -0,0 +1,21 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Tests related to linting of namespace packages."""
+
+from io import StringIO
+
+from pylint.reporters.text import TextReporter
+from pylint.testutils._run import _Run as Run
+
+
+def test_namespace_package_sys_path() -> None:
+    """Test that we do not append namespace packages to sys.path.
+
+    The test package is based on https://github.com/PyCQA/pylint/issues/2648.
+    """
+    pylint_output = StringIO()
+    reporter = TextReporter(pylint_output)
+    Run(["tests/regrtest_data/namespace_import_self/"], reporter=reporter, exit=False)
+    assert "Module import itself" not in pylint_output.getvalue()
```

Comment:
I'm just wondering if we can make this more robust against other fatal errors, as we've seen elsewhere recently. Can we drill into the `.linter.stats` of the `Run` object and assert no fatal errors?

Reply:
I'll have a look!

Reply:
For instance, copying just this test over to `main` passes because of a `syntax-error`. (But the test looks good -- with all of the changes in this PR, minus the substantive changes in `if _is_part_of_namespace_pacakge`, we get the failure with `module-imports-self` as expected.)

Reply:
Let me know what you think of the current solution!

Reply:
Never mind. That test doesn't work 😓

Reply:
The message assertion or the whole thing?

Reply:
At the moment, both 😄 Working on a fix, think I found something!

Reply:
This should be better!